### PR TITLE
feat(anta): Mcs client mounts

### PR DESCRIPTION
**Pull request by sarunac**
# Description

Anta Tests to verify if management cvx mounts for service mcs is in healthy state = mountStateMountComplete

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have run pre-commit for code linting and typing (`pre-commit run`)
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes (`tox -e testenv`)

### CODE REVIEW

File: anta/tests/configuration.py [hunk 1]
```diff
@@ -131,3 +131,42 @@ def test(self) -> None:
...
             self.result.is_failure("Following patterns were not found: " + ",".join(failure_msgs))
+
+
+class VerifyMcsClientMounts(AntaTest):
+    """Verifies the MCS Clients mount status.
+
+    Expected Results
+    ----------------
+    * Success: The test will pass if the MCS mount status on MCS Clients are mountStateMountComplete.
+    * Failure: The test will fail if the MCS mount status on MCS Clients are not mountStateMountComplete.
```
Comment (gmuloc) [inline anta/tests/configuration.py:142]: I think this needs rewording as there are several failure tests and I am not sure what to expect if all clients are mounted or if only one is?

File: anta/tests/configuration.py [hunk 2]
```diff
@@ -131,3 +131,42 @@ def test(self) -> None:
...
+        command_output = self.instance_commands[0].json_output
+        self.result.is_success()
+        mount_states = command_output["mountStates"]
+        mcs_mount_state_detected = False
+        for mount_state in mount_states:
+            if mount_state["type"].startswith("Mcs"):
+                mcs_mount_state_detected = True
+                state = mount_state["state"]
+                if state != "mountStateMountComplete":
+                    self.result.is_failure(f"MCS Client mount states are not valid: {state}")
```
Comment (gmuloc) [inline anta/tests/configuration.py:169]: is this enough information in the error message? would you want to the the type that's failing as well?

Comment (gmuloc) [inline anta/tests/configuration.py:169]:
```suggestion
            if not mount_state["type"].startswith("Mcs"):
                continue
            mcs_mount_state_detected = True
            if (state := mount_state["state"]) != "mountStateMountComplete":
                self.result.is_failure(f"MCS Client mount states are not valid: {state}")
```

File: anta/tests/configuration.py [hunk 3]
```diff
@@ -131,3 +131,42 @@ def test(self) -> None:
...
+
+        if len(mount_states) == 0 or not mcs_mount_state_detected:
```
Comment (gmuloc) [inline anta/tests/configuration.py:171]: this is redundant as if `len(mount_states) == 0` then necessarily `mcs_mount_state_detected` will be False
```suggestion
        if not mcs_mount_state_detected:
```

File: anta/tests/configuration.py [hunk 4]
```diff
@@ -131,3 +131,42 @@ def test(self) -> None:
...
+
+    Expected Results
+    ----------------
+    * Success: The test will pass if the MCS mount status on MCS Clients are mountStateMountComplete.
+    * Failure: The test will fail if the MCS mount status on MCS Clients are not mountStateMountComplete.
+
+    Examples
+    --------
+    ```yaml
+    anta.tests.configuration:
```
Comment (gmuloc) [inline anta/tests/configuration.py:147]: is it really a configuration test?

Would you not prefer a new MCS category maybe?

cc @carl-baillargeon

**Review by gmuloc — CHANGES_REQUESTED** [review]

File: anta/tests/configuration.py (hunk 4 above)
Comment (carl-baillargeon) [reply anta/tests/configuration.py:147]: +1 for having a new category and put the CVX stuff in there.

Comment (gmuloc) [reply anta/tests/configuration.py:147]: let's do it

File: anta/tests/configuration.py [hunk 5]
```diff
@@ -131,3 +131,42 @@ def test(self) -> None:
...
+      - VerifyMcsClientMounts:
+    ```
+    """
+
+    name = "VerifyMcsClientMounts"
+    description = "Verify if all service MCS Client management cvx mounts are mountStateMountComplete"
```
[the remark this answers is not in the document]
Comment (sarunac) [reply anta/tests/configuration.py:153]: Yeah it is verbose! sorry, I will update the description. is there a recommended line limit ?

File: anta/tests/configuration.py (hunk 1 above)
Comment (sarunac) [reply anta/tests/configuration.py:142]: The test will fail even if one of the MCS clients are not in mountStateMountComplete. Should I just add this ?

File: anta/tests/configuration.py (hunk 2 above)
Comment (sarunac) [reply anta/tests/configuration.py:169]: I think this should be enough. There are few other mountStates . But anything other than "complete" is invalid.

File: anta/tests/configuration.py (hunk 4 above)
Comment (sarunac) [reply anta/tests/configuration.py:147]: what would be the new category be? CVXConfiguration ? I do consider it as part of "device configuration"

Comment (carl-baillargeon) [reply anta/tests/configuration.py:147]: You can create a new module --> `anta/tests/cvx.py`

File: anta/tests/cvx.py [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+from typing import TYPE_CHECKING, ClassVar
+
+from anta.models import AntaCommand, AntaTest
+
+if TYPE_CHECKING:
+    from anta.models import AntaTemplate
+
+
+class VerifyMcsClientMounts(AntaTest):
+    """Verifies the MCS Clients mount status.
```
Comment (gmuloc) [inline anta/tests/cvx.py:19]:
```suggestion
    """Verify if all MCS client mounts are in mountStateMountComplete.
```

File: anta/tests/cvx.py [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+    Examples
+    --------
+    ```yaml
+    anta.tests.cvx:
+    - VerifyMcsClientMounts:
+    ```
+    """
+
+    name = "VerifyMcsClientMounts"
+    description = "Verify if all MCS client mounts are in mountStateMountComplete"
```
Comment (gmuloc) [inline anta/tests/cvx.py:35]: We merged a PR that makes these optional - name will be the class name by default and description the first line of the docstring!
```suggestion
```

File: anta/tests/cvx.py [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+    --------
+    ```yaml
+    anta.tests.cvx:
+      - VerifyManagementCVX:
+          enabled: true
+    ```
+    """
+
+    name = "VerifyManagementCVX"
+    description = "Verifies the management CVX global status."
```
Comment (gmuloc) [inline anta/tests/cvx.py:76]: same !
```suggestion
```

**Review by gmuloc — COMMENTED** [review]
will need to add a new section to the doc under `docs/api/test.cvx.md` - you can copy paste any of the other and check (or if you want we can schedule a call to do it together!)

**Comment by sarunac** [comment]
> will need to add a new section to the doc under `docs/api/test.cvx.md` - you can copy paste any of the other and check (or if you want we can schedule a call to do it together!)

done.

**Review by gmuloc — APPROVED** [review]

**Pull request merged**